src/util/texture.cpp

```diff
@@ -1,31 +1,31 @@
 #include "util/texture.h"
 
-QOpenGLTexture* createTexture(const QImage& image) {
+std::unique_ptr<QOpenGLTexture> createTexture(const QImage& image) {
     if (image.isNull()) {
-        return nullptr;
+        return std::unique_ptr<QOpenGLTexture>{};
     }
-    QOpenGLTexture* pTexture = new QOpenGLTexture(image);
+    std::unique_ptr<QOpenGLTexture> pTexture{new QOpenGLTexture(image)};
```

Comment:
We use `make_unique` everywhere. https://stackoverflow.com/a/22571331
```suggestion
    auto pTexture = std::make_unique<QOpenGLTexture>(image);
```

Reply:
Thought we were restricted to c++11

Reply:
Fortunately not. 2.4 is restricted to gcc10, so C++17 is fully okay. On main, you can even use C++20.